public/editor/scripts/editor/js/fc/project-rename.js

```diff
@@ -117,28 +117,31 @@ define(function(require) {
   }
 
   function save(context) {
-    context.saveButton.text("{{ renameProjectSavingIndicator }}");
-
-    persist.call(context, context.titleBar.val(), function(err) {
-      if(err) {
-        console.error("[Bramble] Failed to rename the project with ", err);
-        return;
-      }
+    var contextTitle = context.titleBar.val();
+    // Do not apply save if project title remains the same
+    if (contextTitle === Project.getTitle()) {
+      editingComplete(context);
+    } else {
+      context.saveButton.text("{{ renameProjectSavingIndicator }}");
 
-      Project.setTitle(context.titleBar.val(), function(err) {
+      persist.call(context, contextTitle, function(err) {
         if (err) {
-          console.error("[Bramble] Failed to update the project internally: ", err);
+          console.error("[Bramble] Failed to rename the project. Error: ", err);
           return;
         }
-        editingComplete(context);
-        analytics.event("ProjectRenamed");
 
-        //  Check whether the project is not anonymous		
-        if(context.publisher){
-          context.publisher.showUnpublishedChangesPrompt();		
-        }
+        Project.setTitle(context.titleBar.val(), function(err) {
```

Comment:
we extracted this as `contextTitle` so just use that instead of calling `context.titleBar.val()` a second time.